src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -1206,17 +1196,19 @@ public AuthenticationConfiguration useSaslMechanismProperties(Map<String, ?> mec
     }
 
     /**
-     * Create a new configuration which is the same as this ocnfiguration, but which uses the given kerberos security
+     * Create a new configuration which is the same as this configuration, but which uses the given kerberos security
      * factory to acquire the GSS credential required for authentication.
      *
      * @param kerberosSecurityFactory a reference to the kerberos security factory to be use
      * @return the new configuration
      */
+    @Deprecated
     public AuthenticationConfiguration useKerberosSecurityFactory(SecurityFactory<Credential> kerberosSecurityFactory) {
-        if (Objects.equals(this.kerberosSecurityFactory, kerberosSecurityFactory)) {
-            return this;
+        CredentialSource cs = getCredentialSource();
+        if (kerberosSecurityFactory != null) {
+            return cs != null ? new AuthenticationConfiguration(this, SET_CRED_SOURCE, cs.with(new FactoryCredentialSource(kerberosSecurityFactory))) : new AuthenticationConfiguration(this, SET_CRED_SOURCE, new FactoryCredentialSource(kerberosSecurityFactory));
         }
-        return new AuthenticationConfiguration(this, SET_KRB_SEC_FAC, kerberosSecurityFactory);
+        return this; // cs != null ? new AuthenticationConfiguration(this, SET_CRED_SOURCE, cs.without(GSSKerberosCredential.class)) : this;
```

Comment:
I believe returning `this` in such case is sufficient, as it is consistent with `useSaslMechanismProperties` etc. behavior, as already mentioned.